tableau-databricks/dialect.tdd

```diff
@@ -116,15 +116,187 @@ limitations under the License.
+      <formula>CASE&#10;&#9;WHEN (%1 IS NULL) OR (%2 IS NULL) THEN NULL ELSE GREATEST(%1, %2) END</formula>
+      <argument type='date' />
+      <argument type='date' />
+    </function>
+
+    <function group='numeric' name='MIN' return-type='real'>
+      <formula>CASE&#10;&#9;WHEN (%1 IS NULL) OR (%2 IS NULL) THEN NULL ELSE LEAST(%1, %2) END</formula>
+      <argument type='real' />
+      <argument type='real' />
+    </function>
+    <function group='numeric' name='MIN' return-type='int'>
+      <formula>CASE&#10;&#9;WHEN (%1 IS NULL) OR (%2 IS NULL) THEN NULL ELSE LEAST(%1, %2) END</formula>
+      <argument type='int' />
+      <argument type='int' />
+    </function>
+      <function group='string' name='MIN' return-type='str'>
+      <formula>CASE&#10;&#9;WHEN (%1 IS NULL) OR (%2 IS NULL) THEN NULL ELSE LEAST(%1, %2) END</formula>
+      <argument type='str' />
+      <argument type='str' />
+    </function>
+    <function group='date' name='MIN' return-type='datetime'>
+      <formula>CASE&#10;&#9;WHEN (%1 IS NULL) OR (%2 IS NULL) THEN NULL ELSE LEAST(%1, %2) END</formula>
+      <argument type='datetime' />
+      <argument type='datetime' />
+    </function>
+    <function group='date' name='MIN' return-type='date'>
+      <formula>CASE&#10;&#9;WHEN (%1 IS NULL) OR (%2 IS NULL) THEN NULL ELSE LEAST(%1, %2) END</formula>
+      <argument type='date' />
+      <argument type='date' />
+    </function>
+
+
+    <function group='string' name='MID' return-type='str'>
+      <formula>&#10;SUBSTRING(%1,CAST(%2 AS BIGINT))</formula>
+      <argument type='str' />
+      <argument type='real' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+      <formula>&#10;SUBSTRING(%1,%2)</formula>
+      <argument type='str' />
+      <argument type='int' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+      <formula>&#10;SUBSTRING(%1,CAST(%2 AS INT),CAST(%3 AS BIGINT))</formula>
+      <argument type='str' />
+      <argument type='real' />
+      <argument type='real' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+      <formula>&#10;SUBSTRING(%1,%2,%3)</formula>
+      <argument type='str' />
+      <argument type='int' />
+      <argument type='int' />
+    </function>
+    <function group='string' name='SPACE' return-type='str'>
+      <formula>CASE WHEN CAST(%1 AS BIGINT) &gt;= 0 THEN SPACE(CAST(%1 AS BIGINT)) END</formula>
+      <argument type='real' />
+    </function>
+    <function group='string' name='SPACE' return-type='str'>
+      <formula>CASE WHEN %1 &gt;= 0 THEN SPACE(%1) END</formula>
+      <argument type='int' />
+    </function>
+    <function group='string' name='STARTSWITH' return-type='bool'>
+      <formula>&#10;STARTSWITH(%1, %2)</formula>
+      <argument type='str' />
+      <argument type='str' />
+    </function>
+     <function group='string' name='ENDSWITH' return-type='bool'>
```

Comment:
Are all these functions supported in all DBR 7.3+ releases?

Reply:
We'll only care about 9.1+ release since the earliest train we can catch with Tableau is probably fall. See the design doc for detail test results: https://docs.google.com/document/d/1OoU2VZ_Hi7sQ-ujnhS4BlxsrXvcauiladS_wt_YGKYs/edit

Reply:
Does the tdvt suite have test cases for all newly added functions?

Reply:
Yep, actually the added functions was by inspecting the old TDVT result and making improvements.